src/server/game/Maps/Map.cpp

```diff
@@ -3142,14 +3095,14 @@ bool Map::GetRespawnInfo(RespawnInfoMultiMap const& gridList, RespawnInfoMultiMa
     return false;
 }
 
-void Map::DeleteRespawnInfo(RespawnInfoMultiMap& gridList, RespawnInfoMultiMap& cellAreaZoneList, RespawnInfoMap& spawnList, ObjectGuid::LowType spawnId, uint32 gridId, uint32 cellAreaZoneId, bool onlyDue)
+void Map::DeleteRespawnInfo(RespawnInfoMultiMap& gridList, RespawnInfoMultiMap& zoneList, RespawnInfoMap& spawnList, ObjectGuid::LowType spawnId, uint32 gridId, uint32 zoneId, bool onlyDue)
 {
     // For delete, a bit more important to ensure only one present (or none to erase all)
-    if ((spawnId && (gridId || cellAreaZoneId)) || (gridId && (spawnId || cellAreaZoneId)) || (cellAreaZoneId && (spawnId || gridId)))
-        ASSERT("too many active parameters passed to Map::deleteRespawnInfo");
+    if ((spawnId && (gridId || zoneId)) || (gridId && (spawnId || zoneId)) || (zoneId && (spawnId || gridId)))
+        ASSERT(false && "too many active parameters passed to Map::deleteRespawnInfo");
```

Comment:
Please, ASSERT signature is ASSERT(condition) or ASSERT(condition, msg, ...args), you can use the second version here